### PR TITLE
Users(CUSTOMER) can still be created by the admin

### PR DESCRIPTION
**Pull request by enwemasor1995**
Modified user registration  functionality to be able to register via Backend or online source

### CODE REVIEW

File: graphql/models/User.graphql [hunk 1]
```diff
@@ -5,6 +5,7 @@ type User {
     last_name: String!
     phone_number: String!
     email: String!
+    password:String
```
Comment (omadoyeabraham) [inline graphql/models/User.graphql]: The user password MUST NOT be exposed on the user model. Please remove it.

File: graphql/models/UserProfile.graphql [hunk 2]
```diff
@@ -7,6 +7,7 @@ type UserProfile {
     occupation: String!
     address: String!
     state_of_origin: String!
+    source: Source!
```
Comment (omadoyeabraham) [inline graphql/models/UserProfile.graphql]: Rename this field to registeration_source so it is clear. Make the same change everywhere you previously used the "source" variable

File: graphql/models/UserProfile.graphql [hunk 3]
```diff
@@ -43,6 +44,12 @@ enum Gender {
     FEMALE
 }
 
+# registration source for creating user
+enum Source {
```
Comment (omadoyeabraham) [inline graphql/models/UserProfile.graphql]: Rename this enum to UserRegistrationSource

File: graphql/auth.graphql [hunk 4]
```diff
@@ -52,9 +52,11 @@ input RegisterInput {
     first_name: String! @rules(apply: ["required", "string"])
     last_name: String! @rules(apply: ["required", "string"])
     phone_number: String! @rules(apply: ["required", "string"])
+    password:String
```
Comment (omadoyeabraham) [inline graphql/auth.graphql]: Also add a confirm password field, and add validation to ensure the following:

1. The password and confirm_password fields are the same
2. The password has a minimum of 6 characters.

Read through the lighthouse-php validationn docs [here](https://lighthouse-php.com/4.10/security/validation.html#validating-arguments)

File: resources/views/email/signup/user.blade.php [hunk 5]
```diff
@@ -4,10 +4,18 @@
...
-        * Default Password - {{ $defaultPassword }}
+        @if($source == "BACKEND")
+            {
+            Please find your login credentials below:
+
+            * Email - {{ $user['email'] }}
+            * Default Password - {{ $defaultPassword }}
+            }
+        @else{
+
```
Comment (omadoyeabraham) [inline resources/views/email/signup/user.blade.php]: Where is the text for emails sent to users that register online? 

Note that for online users, do not show their password in the email

File: database/migrations/2019_11_13_073940_create_user_profile_table.php [hunk 6]
```diff
@@ -17,6 +17,7 @@ public function up()
             $table->uuid('id')->unique();
             $table->uuid('user_id');
             $table->string('customer_identifier')->unique();
+            $table->enum('source', ['ONLINE', 'BACKEND']);
```
Comment (omadoyeabraham) [inline database/migrations/2019_11_13_073940_create_user_profile_table.php]: Change to registration_source

File: app/User.php [hunk 7]
```diff
@@ -32,7 +32,7 @@ class User extends Authenticatable
      * @var array
      */
     protected $fillable = [
-        'name', 'email', 'password', 'first_name', 'last_name', 'phone_number'
+        'name', 'email', 'password', 'password_confirmation','first_name', 'last_name', 'phone_number'
```
Comment (omadoyeabraham) [inline app/User.php]: The password confirmation field is not meant to be in the database. It is only for validation

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 8]
```diff
@@ -22,6 +22,7 @@ public function up()
             $table->string('email')->nullable();
             $table->timestamp('email_verified_at')->nullable();
             $table->string('password');
+            $table->string('password_confirmation');
```
Comment (omadoyeabraham) [inline database/migrations/2014_10_12_000000_create_users_table.php]: password confirmation MUST NOT be in the database

File: database/seeds/UsersTableSeeder.php [hunk 9]
```diff
@@ -41,6 +41,7 @@ private function seedSuperAdmin()
             'phone_number' => $faker->phoneNumber,
             'email' => 'admin@admin.com',
             'password' => bcrypt('admin'),
+            'password_confirmation' => bcrypt('admin'),
```
Comment (omadoyeabraham) [inline database/seeds/UsersTableSeeder.php]: Please remove the password confirmation from the DB

File: resources/views/email/signup/user.blade.php [hunk 10]
```diff
@@ -4,11 +4,20 @@
...
 
+            * Email - {{ $user['email'] }}
+            * Default Password - {{ $defaultPassword }}
+            }
+        @else{
+
+        }
+        @endif
+
+        <a href="http://www.springverse.com/login">Login</a>
```
Comment (omadoyeabraham) [inline resources/views/email/signup/user.blade.php]: You need to use the APP_URL environment variable here, and not hardcode a url

Comment (enwemasor1995) [reply resources/views/email/signup/user.blade.php]: Resolved the changes

**Pull request merged**